build/qa/pmd-ruleset.xml

```diff
@@ -85,6 +85,13 @@ under the License.
     <rule ref="category/java/errorprone.xml/UnnecessaryConversionTemporary" />
     <rule ref="category/java/errorprone.xml/UnusedNullCheckInEquals" />
     <rule ref="category/java/errorprone.xml/UselessOperationOnImmutable" />
+    <rule ref="category/java/errorprone.xml/CloseResource" >
+        <properties>
+            <!-- When calling the store to close, PMD wants the full prefix before the call to the method to match, so let's try to use common var names for store ... -->
+            <property name="closeTargets" value="releaseConnection,store.releaseConnection,closeQuietly,closeConnection,closeSafe,store.closeSafe,dataStore.closeSafe,getDataStore().closeSafe,close,closeResultSet,closeStmt"/>
```

Comment:
I am impressed they have this level of configuraiton

Reply:
It's related to how little smart PMD really has to check for proper closing, the above are the templates it's looking for in order to decide something has been actually closed.